pkg/local_object_storage/metabase/delete.go

```diff
@@ -105,16 +96,25 @@ func (db *DB) Delete(prm DeletePrm) (DeleteRes, error) {
 // removed number: objects that were available (without Tombstones, GCMarks
 // non-expired, etc.)
 func (db *DB) deleteGroup(tx *bbolt.Tx, addrs []oid.Address, sizes []uint64) (uint64, uint64, error) {
-	refCounter := make(referenceCounter, len(addrs))
 	currEpoch := db.epochState.CurrentEpoch()
 
 	var rawDeleted uint64
 	var availableDeleted uint64
+	var errorCount uint64
+	var firstErr error
 
 	for i := range addrs {
-		removed, available, size, err := db.delete(tx, addrs[i], refCounter, currEpoch)
+		removed, available, size, err := db.delete(tx, addrs[i], currEpoch)
 		if err != nil {
-			return 0, 0, err // maybe log and continue?
+			errorCount++
+			err = fmt.Errorf("%s object delete fail: %w", addrs[i], err)
```

Comment:
this makes no sense for errors after the 1st: only needed for the log, and it will contain a tautology then. Suggest
```
db.log.Warn("failed to delete object", zap.Stringer("addr", addres[i]), zap.Error(err))
if firstErr == nil {
  err = fmt.Errorf("%s object delete fail: %w", addrs[i], err)
}
```